src/components/table-row.tsx

```diff
@@ -46,7 +52,25 @@ const TableRow = ({
     maxHeightStyle: defaultMaxHeightStyle
   })
 
+  /**
+   * state variable to open and close delete confirmation modal window
+   */
+  const [showDeleteConfirmationModal, setShowDeleteConfirmationModal] = useState<boolean>(false);
```

Comment:
Instead of a boolean, you should have an object that if it's defined, it will be the props of delete-confirmation modal.

The props would be,
- The tuple that will be deleted.
- The confirm callback. This callback will be different if 'isUnlink' is true, that's why I suggest storing the callback in the state. So you don't have to store an extra `isUnlink` variable.